application/helpers/surveyThemeHelper.php

```diff
@@ -0,0 +1,431 @@
+<?php if (!defined('BASEPATH')) {
+    exit('No direct script access allowed');
+}
+/*
+* LimeSurvey
+* Copyright (C) 2007-2021 The LimeSurvey Project Team / Carsten Schmitz
+* All rights reserved.
+* License: GNU/GPL License v2 or later, see LICENSE.php
+* LimeSurvey is free software. This version may have been modified pursuant
+* to the GNU General Public License, and as distributed it includes or
+* is derivative of works licensed under the GNU General Public License or
+* other free or open source software licenses.
+* See COPYRIGHT.php for copyright notices and details.
+*
+*/
+
+use LimeSurvey\Datavalueobjects\ThemeFileCategory;
+use LimeSurvey\Datavalueobjects\ThemeFileInfo;
+
+/**
+ * General helper class for survey themes
+ */
+class surveyThemeHelper
+{
+    /** @var array $aTemplatesInUploadDir cache for the method getTemplateInUpload */
+    public static $aTemplatesInUploadDir = null;
```

Comment:
No Hungarian notation in new files.

Also, feel free to be more specific than "array", e.g. "string[]" for a list of strings.